src/Runner.Worker/ActionManager.cs

```diff
@@ -1068,8 +928,21 @@ private ActionContainer PrepareRepositoryActionAsync(IExecutionContext execution
                 }
                 else if (actionDefinitionData.Execution.ExecutionType == ActionExecutionType.Composite)
                 {
-                    Trace.Info($"Action composite: {(actionDefinitionData.Execution as CompositeActionExecutionData).Steps}, no more preparation.");
-                    return null;
+                    // TODO: we need to generate unique Id's for composite steps
+                    Trace.Info($"Loading Composite steps");
+                    var compositeAction = actionDefinitionData.Execution as CompositeActionExecutionData;
+                    setupInfo.Steps = compositeAction.Steps;
+
+                    foreach (var step in compositeAction.Steps)
+                    {
+                        step.Id = Guid.NewGuid();
```

Comment:
+1 looks like this isn't used for ExecutionContext.CreateEmbeddedChild but i believe this is needed for tracking "intra-state" (pre/main/post)

Reply:
wonder if we have a bug today where all nested "main" share the same intra-state dictionary